sbom/update_component_sbom.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+"""
+This script updates the purls in component-level SBOMs with release time info.
+"""
+import argparse
+import glob
+import json
+import os
+from collections import defaultdict
+from typing import DefaultDict, Dict, List
+
+
+def get_component_to_purls_map(images: List[Dict]) -> Dict[str, List[str]]:
+    """
+        Get dictionary mapping component names to list of image purls.
+
+    Args:
+        images: List of image metadata from the given data.json.
+
+    Returns:
+        Dictionary mapping of component names to list of purls.
+    """
+    component_purls: DefaultDict[str, List[str]] = defaultdict(list)
+
+    for image in images:
+        component = image["component"]
+        purl = image["purl"]
+        component_purls[component].append(purl)
+
+    return dict(component_purls)
+
+
+def update_cyclonedx_sbom(sbom: Dict, component_to_purls_map: Dict[str, List[str]]) -> None:
+    """
+    Update the purl in an SBOM with CycloneDX format
+    Args:
+        sbom: CycloneDX SBOM file to update.
+        component_to_purls_map: dictionary mapping of component names to list of purls.
+    """
+    for component in sbom["components"]:
+        if component["name"] in component_to_purls_map:
+            # only one purl is supported for CycloneDX
+            component["purl"] = component_to_purls_map[component["name"]][0]
+
+
+def update_spdx_sbom(sbom: Dict, component_to_purls_map: Dict[str, List[str]]) -> None:
+    """
+    Update the purl in an SBOM with SPDX format
+    Args:
+        sbom: SPDX SBOM file to update.
+        component_to_purls_map: dictionary mapping of component names to list of purls.
+    """
+    for package in sbom["packages"]:
+        if package["name"] in component_to_purls_map:
+            purl_external_refs = [
+                {
+                    "referenceCategory": "PACKAGE-MANAGER",
+                    "referenceType": "purl",
+                    "referenceLocator": purl,
+                }
+                for purl in component_to_purls_map[package["name"]]
+            ]
+            package["externalRefs"].extend(purl_external_refs)
+
+
+def update_sboms(data_path: str, input_path:str, output_path:str) -> None:
+    """
+    Update all SBOM files in the given input_path directory, and save the updated SBOMs to the
+    output_path directory
+    Args:
+        data_path: path to data.json file containing image metadata.
+        input_path: path to directory holding SBOMs files to be updated.
+        output_path: path to directory to save updated SBOMs.
+    """
+    with open(data_path, "r") as data_file:
+        data = json.load(data_file)
+        release_notes = data["releaseNotes"]
+
+    component_to_purls_map = get_component_to_purls_map(release_notes["images"])
```

Comment:
Technically, only the `json.load` file needs the open file, right? And also, the rest of `release_notes` is never used, only images. So maybe you can simplify it like this?
```suggestion
    with open(data_path, "r") as data_file:
        data = json.load(data_file)

    component_to_purls_map = get_component_to_purls_map(data["release_notes"]["images"])
```